docs/crr-pause-resume.md

```diff
@@ -0,0 +1,54 @@
+# Cross-Region Replication (CRR) Manual Pause and Resume
+
+## Description
+
+This feature offers a way for users to manually pause and resume cross-region
+replication (CRR) operations by site.
+
+## Design
+
+A RESTful API will expose methods for users to pause and resume cross-region
+replication operations.
+
+Redis offers a pub/sub function. We will utilize this function to propagate
+requests to all active CRR Kafka Consumers on all nodes with backbeat
+containers setup for replication.
+
+We want to pause and resume the CRR service at the lowest level (in our case,
+pause and resume all Kafka Consumers subscribed to the CRR topic). We want to
+perform these actions at the lowest level in order to stop processing any
+current batch jobs.
+
+The API will have a Redis instance publishing messages to a specific channel.
+All CRR Kafka Consumers will subscribe to this channel and complete any given
+valid requests.
+
+When a Kafka Consumer pauses, the Consumer is still kept alive and maintains
+any internal state, including offset. The Consumer will no longer be
+subscribed to the CRR topic, so will no longer try consuming any entries.
+When the paused Consumer is resumed, it will again resume consuming entries
+from its last offset.
+
+## Definition of API
+
+* POST `/_/crr/pause/<site>`
+
+    This POST request is to manually pause the cross-region replication service.
+    You may specify a site name, or if you would like to pause replication for
+    all sites, specify "all" as the site name.
+
+    Response:
+    ```sh
+    {}
```

Comment:
It might be helpful UX to include some kind of confirmation in the response. I'm imagining something like:

```
{
    aws-backend: PAUSED,
}
```

Reply:
i'm a little hesitant in returning the status right away because when a user sends a request to pause or resume, I don't confirm that it was actually set. Since i'm using Redis to communicate between the processes, if the subscriber fails in receiving the message, there is no way for the publisher to know that unless it checks crr status (which will be opened in a separate pr)

Instead, I'm hoping a user using this feature will perform the pause/resume action and will send a separate request to check status. That way, the info provided to the user will be more accurate